### PR TITLE
Add fp

### PR DESCRIPTION
**Pull request by piazzatron**
### Dragons

None

### How Has This Been Tested?

- Tested on stage

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### CODE REVIEW

File: packages/web/src/services/fingerprint/FingerprintClient.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    console.log('Initializing Fingerprint client')
+    try {
+      const fp = await FingerprintJS.load({
+        apiKey: this.apiKey,
+        endpoint: this.endpoint
+      })
+      console.log(`Fingerprint loaded`)
+      this.fingerprint = fp
+    } catch (e) {
+      console.error(`Error initializaing fingerprint client: ${e}`)
```
Comment (raymondjacobson) [inline packages/web/src/services/fingerprint/FingerprintClient.ts:28]: initializing

File: packages/web/src/services/fingerprint/FingerprintClient.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  async identify(userId: number) {
+    if (!this.fingerprint) {
+      console.warn('Fingerprint client not yet initted')
+      return
+    }
+    try {
+      const origin = isMobile() ? 'mobile' : isElectron() ? 'desktop' : 'web'
+
+      // First, see if we've fingerprinted this user before
+      const response = await fetch(
```
Comment (raymondjacobson) [inline packages/web/src/services/fingerprint/FingerprintClient.ts:41]: Non-blocking, but we could probably avoid an extra network req if we remember in local storage that we have done a fprint

**Review by raymondjacobson — APPROVED** [review]

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/piazz-fingerprint

**Review by sddioulde — APPROVED** [review]
lg

File: packages/web/src/services/fingerprint/FingerprintClient.ts (hunk 2 above)
Comment (piazzatron) [reply packages/web/src/services/fingerprint/FingerprintClient.ts:41]: interesting idea - would be a bit problematic in the case that you switch browsers / use incognito, and we'd end up fingerprinting multiple times. going to leave as if for now

Comment (raymondjacobson) [reply packages/web/src/services/fingerprint/FingerprintClient.ts:41]: don't we want to fp again if you switch browsers?

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/piazz-fingerprint

File: packages/web/src/services/fingerprint/FingerprintClient.ts (hunk 2 above)
Comment (piazzatron) [reply packages/web/src/services/fingerprint/FingerprintClient.ts:41]: discussed online – fp is stable between browsers

**Pull request merged**